File: backend-service/src/main/java/ch/micha/automation/room/scene/dtos/SceneDTO.java
package ch.micha.automation.room.scene.dtos;

import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class SceneDTO {
    private int id;
    private String name;
    private boolean defaultScene;
    private String spotifyResource;
    private int spotifyVolume;
    private List<SceneLightConfigDTO> lights;
}

File: admin-ui/src/app/dtos/scene/SceneDTO.ts
import {SceneLightConfigDTO} from "./SceneLightConfigDTO";
import {SpotifyResourceDTO} from "../spotify/SpotifyResourceDTO";

export type SceneDTO = {
  id: number,
  name: string,
  defaultScene: boolean,
  spotifyResource: SpotifyResourceDTO,
  spotifyVolume: number,
  lights: SceneLightConfigDTO[],
};

export type SimpleSceneDTO = {
  id: number,
  name: string,
  defaultScene: boolean,
  spotifyResourceName: string,
  spotifyResourceImage: string,
  spotifyVolume: number,
  deviceCount: number,
};